ipaserver/install/ipa_backup.py

```diff
@@ -358,6 +363,79 @@ def run(self):
                 logger.error('Cannot change directory to %s: %s', cwd, e)
             shutil.rmtree(self.top_dir)
 
+    def check_roles(self, raiseonerr=True):
+        """Check that locally-installed roles match the globally used ones.
+
+        Specifically: make sure no role used in the cluster is absent
+        from the local replica ipa-backup is running on.
+        """
+
+        locally_installed_roles = set()
+        globally_used_roles = set()
+
+        # We need to cover the following roles:
+        # * DNS: filter="(|(cn=DNS)(cn=DNSKeySync))"
+        # * CA:  filter="(cn=CA)"
+        # * KRA: filter="(cn=KRA)"
+        # * AD Trust Controller: filter="(cn=ADTRUST)"
+        # Note:
+        # We do not need to worry about AD Trust Agents as Trust
+        # Controllers are Trust Agents themselves and contain extra,
+        # necessary Samba configuration. So either the cluster has no
+        # AD Trust bits installed, or it should be backuped on a Trust
+        # Controller, not a Trust Agent.
+        role_names = {
+            'CA', 'DNS', 'DNSKeySync', 'KRA', 'ADTRUST'
+        }
+
+        search_base = DN(api.env.container_masters, api.env.basedn)
+        attrs_list = ['ipaConfigString', 'cn']
```

Comment:
Please make the attributes all lower-case.

Reply:
I think this is resolved now.